packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -828,6 +826,7 @@ export const DatePickerMixin = (subclass) =>
     /** @protected */
     _onOverlayOpened() {
       const content = this._overlayContent;
+      content.reset();
```

Comment:
Moved this logic from `connectedCallback()` to the separate method in order to work with Lit.